### PR TITLE
LOG-974: Fix username password for ES

### PR DESCRIPTION
**Pull request by igor-karpukhin**
### Description
This PR adds proper username/password handling for the ES

/cc @jcantrill 
/assign @jcantrill 

### Links
- JIRA: https://issues.redhat.com/browse/LOG-974

### CODE REVIEW

File: pkg/generators/forwarding/fluentd/fluent_conf.go [hunk 1]
```diff
@@ -113,7 +113,46 @@ func (conf *outputLabelConf) BufferPath() string {
...
+	}
+	return true
+}
+
+func (conf *outputLabelConf) HasTLSKeyAndCrt() bool {
+	if conf.Secret == nil {
+		return false
+	}
+
+	if _, ok := conf.Secret.Data["tls.key"]; !ok {
```
Comment (jcantrill) [inline pkg/generators/forwarding/fluentd/fluent_conf.go:138]: I believe `constants.go` has a value for this and the other opinionated secrets

File: pkg/generators/forwarding/fluentd/fluent_conf_test.go [hunk 2]
```diff
@@ -15,12 +17,55 @@ var _ = Describe("Generating fluentd config", func() {
...
 		var err error
 		generator, err = NewConfigGenerator(false, false, true)
 		Expect(err).To(BeNil())
 		Expect(generator).ToNot(BeNil())
+		secrets = map[string]*corev1.Secret{
+			"infra-es": {
+				ObjectMeta: v1.ObjectMeta{
+					Name: "my-infra-secret",
+				},
+				Data: map[string][]byte{
```
Comment (jcantrill) [inline pkg/generators/forwarding/fluentd/fluent_conf_test.go:32]: nit... one could move this map into a single var for reuse...

File: pkg/k8shandler/forwarding.go [hunk 3]
```diff
@@ -43,6 +43,9 @@ func (clusterRequest *ClusterLoggingRequest) generateCollectorConfig() (config s
 				// copy from defaults if output specific spec not present
 				if out.Type == logging.OutputTypeElasticsearch && out.Elasticsearch == nil {
 					out.Elasticsearch = defaultOutputSpecs.Elasticsearch
+					out.Secret = &logging.OutputSecretSpec{
+						Name: "fluentd",
```
Comment (jcantrill) [inline pkg/k8shandler/forwarding.go:47]: We MAY have a constant for the secret name too

**Review by jcantrill — CHANGES_REQUESTED** [review]
@igor-karpukhin  a few nits, but overall looks good.  Please squash commits

File: pkg/generators/forwarding/fluentd/fluent_conf.go (hunk 1 above)
Comment (igor-karpukhin) [reply pkg/generators/forwarding/fluentd/fluent_conf.go]: Fixed

File: pkg/generators/forwarding/fluentd/fluent_conf_test.go (hunk 2 above)
Comment (igor-karpukhin) [reply pkg/generators/forwarding/fluentd/fluent_conf_test.go]: Fixed

File: pkg/k8shandler/forwarding.go (hunk 3 above)
Comment (igor-karpukhin) [reply pkg/k8shandler/forwarding.go]: Fixed

**Pull request closed without merging**